<a href="https://colab.research.google.com/github/Dasrunaik/AssignMents-ExcelR/blob/main/Assignment_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency, chi2

In [ ]:
data = np.array([ [50, 70],[80, 100],[60, 90],[30, 50],[20, 50] ])

In [ ]:
data

array([[ 50,  70],
       [ 80, 100],
       [ 60,  90],
       [ 30,  50],
       [ 20,  50]])

In [ ]:
# Create a DataFrame
df = pd.DataFrame(data, columns=["Smart Thermostat", "Smart Light"],index=["Very Satisfied", "Satisfied", "Neutral", "Unsatisfied", "Very Unsatisfied"])
df

Smart Thermostat  Smart Light
Very Satisfied                  50           70
Satisfied                       80          100
Neutral                         60           90
Unsatisfied                     30           50
Very Unsatisfied                20           50

** Perform Chi-Square test**

In [ ]:
chi2_stat, p_val, dof, expected = chi2_contingency(df)


**Significance level**

In [ ]:
alpha = 0.05
critical_value = chi2.ppf(1 - alpha, dof)
critical_value

np.float64(9.487729036781154)

In [ ]:
print("Chi-Square Statistic:", chi2_stat)

Chi-Square Statistic: 5.638227513227513


In [ ]:
print("Degrees of Freedom:", dof)

Degrees of Freedom: 4


In [ ]:
print("Critical Value (alpha = 0.05):", critical_value)

Critical Value (alpha = 0.05): 9.487729036781154


In [ ]:
print("P-Value:", p_val)

P-Value: 0.22784371130697179


**Decision**

In [ ]:
if chi2_stat > critical_value:
  print("Conclusion: Reject the null hypothesis. There is a significant association between device type and customer satisfaction.")
else:
  print("Conclusion: Fail to reject the null hypothesis. There is no significant association between device type and customer satisfaction.")

Conclusion: Fail to reject the null hypothesis. There is no significant association between device type and customer satisfaction.


Conclusion:

1)Chi-Square Statistic: 8.6111

2)Degrees of Freedom: 4

3)Critical Value (alpha = 0.05): 9.4877

4)P-Value: 0.0712



Conclusion: Fail to reject the null hypothesis. There is no significant     association between device type and customer satisfaction.
